diff --git a/dfp.admin.inc b/dfp.admin.inc
index ce0f01f..f4e0d9e 100644
--- a/dfp.admin.inc
+++ b/dfp.admin.inc
@@ -238,6 +238,14 @@ function dfp_admin_settings($form, $form_state) {
     '#default_value' => variable_get('dfp_injected_js2', ''),
     '#rows' => 5,
   );
+  $form['javascript']['dfp_injected_js3'] = array(
+    '#type' => 'textarea',
+    '#title' => t('Inject javascript') . ' 3',
+    '#description' => t('Inject this javascript into the @tag on every page request inside of the googletag.cmd.push function
+.', array('@tag' => '<head>')),
+    '#default_value' => variable_get('dfp_injected_js3', ''),
+    '#rows' => 5,
+  );
 
   return system_settings_form($form);
 }
diff --git a/dfp.module b/dfp.module
index 55acbfa..cbbefa3 100755
--- a/dfp.module
+++ b/dfp.module
@@ -687,6 +687,7 @@ function _dfp_js_global_settings() {
   // Add global settings with a heavy weight so that they appear after all the
   // defineSlot() calls otherwise IE8 and Opera fail to display ads properly.
   $js = 'googletag.cmd.push(function() {' . "\n";
+  $js .= variable_get('dfp_injected_js3', '') . "\n";
   if (variable_get('dfp_async_rendering', 1)) {
     $js .= '  googletag.pubads().enableAsyncRendering();' . "\n";
   }
